src/components/Identity/createComponent.js

```diff
@@ -61,14 +55,23 @@ export default (processIdSyncs, config, logger, consent, eventManager) => {
         // new visitor).
         deferredForEcid = defer();
         payload.expectResponse();
-        // If third-party cookies are enabled by the customer and
-        // supported by the browser, we will send the request to a
-        // a third-party domain that allows for more accurate
-        // identification of the user through use of a third-party cookie.
-        if (
+
+        if (ecidToMigrate) {
+          // We don't have an identity cookie, but we do have an ECID
+          // from a legacy cookie that we can explicitly provide
+          // to the server. Since we have an ECID, it might be reasonable to
+          // allow other requests to go out in parallel, but to make sure
+          // there aren't unforeseen problems, we will hold back other
+          // requests until the identity cookie gets set.
+          addEcidToPayload(payload, ecidToMigrate);
+        } else if (
           config.thirdPartyCookiesEnabled &&
           areThirdPartyCookiesSupportedByDefault(getBrowser(window))
         ) {
+          // If third-party cookies are enabled by the customer and
```

Comment:
Can you add a comment here explaining why not to use the third party domain if we already have an ecid?

Reply:
Yes. My reasoning was that we only hit demdex when we need an ECID value and can use the demdex cookie to help retrieve that ECID. Since we already have an ECID, I don't think hitting demdex would provide any value. I sent Rares an additional message just to verify that that's the case. Maybe @jfkhoury can help verify. I'll add a comment though.

Reply:
I talked to Rares about this some more. He said that if we have an ECID to migrate, it would probably still be a good idea to hit demdex because if the demdex cookie isn't already set, demdex will use our ECID to set the demdex cookie. I've made the change and updated the comments to hopefully be clearer about why we're doing the things we're doing.